### ISSUE TITLE
Choose: SVG or Canvas?

### ISSUE THREAD

**Issue by girum11** [tessa3/uxgraph#20]

**Comment by eyuelt** [comment]
**tl;dr**: Going with SVG.

SVG is better in this case because we want to be able to interact with the elements. In canvas we would have to reimplement a lot to be able to enable this, as canvas is pixel-based and doesn't keep track of the stuff drawn on it.

Benefits of SVG:
- SVG elements are in the DOM.
 - their contents are highlightable and interactable
 - they are inspectable in the browser dev tools, which makes debugging much easier
 - they can have event listeners just like any other DOM element, while canvas requires reimplementing event handling and determining which elements get which events
- SVGs are vector images.
 - they scale well regardless of zoom level, device DPI, etc.
- Sketch exports SVGs.
- SVG supports CSS.

Other considerations:
- We won't need to redraw super often, which would be a good use case for canvas.
- We likely won't have anywhere near enough elements to give the browser trouble, which would be a good argument for canvas, as it doesn't add elements to the DOM.
